dao/src/main/java/uk/gov/moj/sdt/dao/IndividualRequestDao.java

```diff
@@ -1,176 +1,202 @@
+            Predicate[] sdtCustomerPredicate = createCriteria(sdtReferenceId);
+            return criteriaQuery.select(root).where(sdtCustomerPredicate);
+        });
+
+        // Should only return one or none at all
+        if (individualRequests == null || individualRequests.length == 0) {
+            return null;
+        }
+
+        if (individualRequests.length > 1) {
+            throw new IllegalStateException("Multiple Individual Requests found for the Sdt Request Reference " +
+                    sdtReferenceId);
+        }
+
+        return (IIndividualRequest) individualRequests[0];
+    }
+
+    @Override
+    public List<IIndividualRequest> getPendingIndividualRequests(final int maxAllowedAttempts)
+            throws DataAccessException {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Get pending individual requests in status FORWARDED that have  "
+                    + "reached their maximum forwarding attempts.");
+        }
+
+        return this.queryAsList(IIndividualRequest.class, () -> {
+            Predicate[] sdtCustomerPredicate =  createCriteria(IndividualRequest.IndividualRequestStatus.FORWARDED.getStatus(),
+                                                               maxAllowedAttempts, false);
+            return criteriaQuery.select(root).where(sdtCustomerPredicate);
+        });
+    }
+
+    @Override
+    public List<IIndividualRequest> getStaleIndividualRequests(final int minimumAgeInMinutes)
+            throws DataAccessException {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Get unforwarded individual requests created more than " + minimumAgeInMinutes +
+                    " minutes ago");
+        }
+
+        final LocalDateTime now = LocalDateTime.now();
+        final LocalDateTime latestTime = now.minusMinutes(minimumAgeInMinutes);
+
+        return this.queryAsList(IndividualRequest.class, () -> {
+            Predicate[] sdtCustomerPredicate =   createCriteria(false, latestTime);
+            return criteriaQuery.select(root).where(sdtCustomerPredicate);
+        });
+    }
+
+    private Predicate[] createCriteria(boolean deadLetter, LocalDateTime latestTime) {
+        Predicate[] predicates = new Predicate[3];
+        predicates[0] = criteriaBuilder.or(criteriaBuilder.equal(root.get("requestStatus"), IIndividualRequest.IndividualRequestStatus.RECEIVED.getStatus()),
+                                           criteriaBuilder.equal(root.get("requestStatus"), IIndividualRequest.IndividualRequestStatus.FORWARDED.getStatus()));
+        predicates[1] = criteriaBuilder.equal(root.get("deadLetter"), deadLetter);
+        predicates[2] = criteriaBuilder.or(criteriaBuilder.lessThan(root.get("updatedDate"), latestTime), criteriaBuilder.isNull(root.get("updatedDate")));
+        return predicates;
+    }
+
+    private Predicate[] createCriteria(String sdtRequestReference) {
```

Comment:
Might be better to name the createCriteria() methods after their purpose, e.g. createCriteriaSdtRequestRef() and createCriteriaStaleIndividualRequest().  The methods create criteria for specific purposes and renaming them makes clear what that is.  It will also make it easier to understand the code.

Reply:
rename to createCriteria  to createSdtRequestReferencePredicate and createIndividualRequestPredicate